core/application/commands.go

```diff
@@ -26,17 +26,25 @@ func (a *app) CreateMicroVM(ctx context.Context, mvm *models.MicroVM) (*models.M
 			return nil, fmt.Errorf("generating random name for microvm: %w", err)
 		}
 
-		vmid, err := models.NewVMID(name, defaults.MicroVMNamespace)
+		vmid, err := models.NewVMID(name, defaults.MicroVMNamespace, "")
 		if err != nil {
 			return nil, fmt.Errorf("creating vmid: %w", err)
 		}
 
 		mvm.ID = *vmid
 	}
 
+	uid, err := a.ports.IdentifierService.GenerateRandom()
+	if err != nil {
+		return nil, fmt.Errorf("generating random name for microvm: %w", err)
```

Comment:
the nittiest of nits:
```suggestion
		return nil, fmt.Errorf("generating random ID for microvm: %w", err)
```